signer/src/context/signer_state.rs

```diff
@@ -537,6 +537,18 @@ impl SignerSet {
             .expect("BUG: Failed to acquire read lock")
             .len()
     }
+
+    /// Returns true if the two signer sets have the same public keys.
+    pub fn has_same_pubkeys<I, P>(&self, other: I) -> bool
+    where
+        I: IntoIterator<Item = P>,
+        P: Into<PublicKey>,
```

Comment:
No need to make this generic, since we only use `BTreeSet<PublicKey>` for the type of `other`.

Reply:
Are you sure we will never want to use it for HashSet or Vec ? And also I think this generic don't look very complex.

Reply:
Yeah, it's not that complex. It's just that it adds unnecessary complexity; we need to add even more code to satisfy the trait bounds (we needed the `From<&PublicKey>` implementation). Instead we should skip all of that and just do what we need to. Most of our functions are not generic unless they reduce complexity (well, that's the goal at least).

Maybe we never need to use this with a `HashSet` or whatever, or we will. We can always change it in the future.

Reply:
[ef75378](https://github.com/stacks-sbtc/sbtc/pull/1602/commits/ef7537847ffe636587abe7e733399f34bf95adc7)